controllers/hcloudmachinetemplate_controller.go

```diff
@@ -65,6 +66,13 @@ func (r *HCloudMachineTemplateReconciler) Reconcile(ctx context.Context, req rec
 
 	// Fetch the Cluster.
 	cluster, err := util.GetOwnerCluster(ctx, r.Client, machineTemplate.ObjectMeta)
+	if cluster == nil {
```

Comment:
maybe make the call for clusterClass first and then if you get an error, then you get an error you just ignore it and try GetOwnerCluster